### PR TITLE
Add support for PEX runtime tools & an info tool.

### PR DESCRIPTION
**Pull request by jsirois**
Add a new `--include-tools` option to include any pex.tools in generated
PEX files. These tools are activated by running PEX files with
PEX_TOOLS=1. The `Info` tool seeds the tool set and simply dumps the
effective PEX-INFO for the given PEX.

Work towards #962 and #1115

### CODE REVIEW

**Comment by jsirois** [comment]
N.B.: Graph, Interpreter and Venv tools are completed and staged for fast-follow PRs to add to the tool set.

An example for the current toolset using a `fabric` PEX built with:
```
$ python -mpex -c fab fabric -ofabric.pex --include-tools
```

Use tools (incorrectly):
```
$ PEX_TOOLS=1 ./fabric.pex
usage: PEX_TOOLS=1 ./fabric.pex [-h] {info} ...
PEX_TOOLS=1 ./fabric.pex: error: a subcommand is required
```

Find out more:
```
$ PEX_TOOLS=1 ./fabric.pex -h
usage: PEX_TOOLS=1 ./fabric.pex [-h] {info} ...

Tools for working with ./fabric.pex.

optional arguments:
  -h, --help  show this help message and exit

subcommands:
  The PEX file ./fabric.pex can be operated on using any of the following subcommands.

  {info}
    info      Dumps the PEX-INFO json contained in a PEX file.
```

And drill down to the `info` tool:
```
$ PEX_TOOLS=1 ./fabric.pex info -h
usage: PEX_TOOLS=1 ./fabric.pex info [-h] [-o PATH] [-i INDENT]

Dumps the PEX-INFO json contained in a PEX file.

optional arguments:
  -h, --help            show this help message and exit
  -o PATH, --output PATH
                        A file to output the PEX-INFO json to; STDOUT by default or when `-` is specified.
  -i INDENT, --indent INDENT
                        Pretty-print PEX-INFO json with the given indent.
```

Then use the tool:
```
$ PEX_TOOLS=1 ./fabric.pex info -i4
{
    "always_write_cache": false,
    "build_properties": {
        "class": "CPython",
        "pex_version": "2.1.21",
        "platform": "manylinux2014_x86_64",
        "version": [
            3,
            9,
            0
        ]
    },
    "code_hash": "da39a3ee5e6b4b0d3255bfef95601890afd80709",
    "distributions": {
        "PyNaCl-1.4.0-cp35-abi3-manylinux1_x86_64.whl": "bf72b2b403888b38d073f7f149499715d6bce08c",
        "bcrypt-3.2.0-cp36-abi3-manylinux2010_x86_64.whl": "735f27eb7ad8b8921f89efcebfc33ed7bac8f465",
        "cffi-1.14.4-cp39-cp39-manylinux1_x86_64.whl": "203a578371ccb66782b17a75b0f574200a57f210",
        "cryptography-3.2.1-cp35-abi3-manylinux2010_x86_64.whl": "e2b94725fb6135fbf52d3d5c34ce57a3c6236c43",
        "fabric-2.5.0-py2.py3-none-any.whl": "2113e91016146d06b9c050838033373500bc905b",
        "invoke-1.4.1-py3-none-any.whl": "1f021853a33baccac61ffe9ebf6f16a2e5ef357f",
        "paramiko-2.7.2-py2.py3-none-any.whl": "803260f2de7a361df72d417ba064851a90c99226",
        "pycparser-2.20-py2.py3-none-any.whl": "4bf5fb0f6540ca05cb505b5bba08c37043075a90",
        "six-1.15.0-py2.py3-none-any.whl": "0cb007434fc9c1d2ee600dbd964088b69f5d670a"
    },
    "emit_warnings": true,
    "entry_point": "fabric.main:program.run",
    "ignore_errors": false,
    "inherit_path": "false",
    "interpreter_constraints": [],
    "pex_path": null,
    "requirements": [
        "PyNaCl==1.4.0; python_full_version != \"3.0.*\" and python_full_version != \"3.1.*\" and python_full_version != \"3.2.*\" and python_full_version != \"3.3.*\" and python_version >= \"2.7\"",
        "bcrypt==3.2.0; python_version >= \"3.6\"",
        "cffi==1.14.4",
        "cryptography==3.2.1; python_full_version != \"3.0.*\" and python_full_version != \"3.1.*\" and python_full_version != \"3.2.*\" and python_full_version != \"3.3.*\" and python_full_version != \"3.4.*\" and python_version >= \"2.7\"",
        "fabric==2.5.0",
        "invoke==1.4.1",
        "paramiko==2.7.2",
        "pycparser==2.20; python_full_version != \"3.0.*\" and python_full_version != \"3.1.*\" and python_full_version != \"3.2.*\" and python_full_version != \"3.3.*\" and python_version >= \"2.7\"",
        "six==1.15.0; python_full_version != \"3.0.*\" and python_full_version != \"3.1.*\" and python_full_version != \"3.2.*\" and python_version >= \"2.7\""
    ],
    "strip_pex_env": true,
    "unzip": false,
    "zip_safe": true
}
```

For anything fancier, folks can use `jq` to drill down.

**Comment by jsirois** [comment]
Sorry for the patch thrash - I was doing some git gymnastics and fell. Good to go now.

File: pex/pex.py [hunk 1]
```diff
@@ -436,19 +449,37 @@ def execute(self):
...
             working_set = self._activate()
             self.patch_pkg_resources(working_set)
-            exit_code = self._wrap_coverage(self._wrap_profiling, self._execute)
+            if self._vars.PEX_TOOLS:
+                try:
+                    from pex.tools import main as tools
+                except ImportError as e:
+                    die(
+                        "This PEX was not built with tools (Re-build the PEX file with "
+                        "`pex --include-tools ...`): {}".format(e)
```
Comment (stuhood) [inline pex/pex.py:472]: Maybe worth mentioning why this was triggered.

`"The PEX_TOOLS environment variable was set, but this PEX was not ..."`

**Review by stuhood — APPROVED** [review]
Looks good!

Added a comment on #1115 to discuss the direction a bit.

File: pex/pex.py (hunk 1 above)
Comment (jsirois) [reply pex/pex.py:472]: Agreed. I'll hit that in one of the follow-up tools PRs here presently to save some trees.

**Pull request merged**